### PR TITLE
IOS/Network/IP/Top: Separate behaviors into their own functions

### PR DESCRIPTION
**Pull request by lioncash**
Localizes behavior and separates the code out from the IOCtl/IOCtlV switch blocks.

### CODE REVIEW

File: Source/Core/Core/IOS/Network/IP/Top.cpp [hunk 1]
```diff
@@ -155,48 +155,17 @@ IPCCommandResult NetIPTop::IOCtl(const IOCtlRequest& request)
...
   case IOCTL_SO_ICMPCLOSE:
-  {
-    u32 fd = Memory::Read_U32(request.buffer_in);
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    return_value = sm.DeleteSocket(fd);
-    INFO_LOG(IOS_NET, "%s(%x) %x",
-             request.request == IOCTL_SO_ICMPCLOSE ? "IOCTL_SO_ICMPCLOSE" : "IOCTL_SO_CLOSE", fd,
-             return_value);
-    break;
-  }
```
Comment (lioncash) [inline Source/Core/Core/IOS/Network/IP/Top.cpp]: GitHub won't let me directly put a comment on it, but I wasn't sure what to name the function for the code following this comment (for accept, bind, connect, and fcntl). If you have a naming suggestion, feel free to let me know and I'll move this code into a function with that name.

Comment (Parlane) [inline Source/Core/Core/IOS/Network/IP/Top.cpp]: Do you mean DoSock ?

Comment (lioncash) [inline Source/Core/Core/IOS/Network/IP/Top.cpp]: yeah, the following:

```cpp
case IOCTL_SO_ACCEPT:
case IOCTL_SO_BIND:
case IOCTL_SO_CONNECT:
case IOCTL_SO_FCNTL:
{
 u32 fd = Memory::Read_U32(request.buffer_in);
 WiiSockMan& sm = WiiSockMan::GetInstance();
 sm.DoSock(fd, request, static_cast<NET_IOCTL>(request.request));
 return GetNoReply();
}
```

File: Source/Core/Core/IOS/Network/IP/Top.cpp [hunk 2]
```diff
@@ -155,802 +155,844 @@ IPCCommandResult NetIPTop::IOCtl(const IOCtlRequest& request)
...
+  case IOCTL_SO_SETSOCKOPT:
+    return SetSockOpt(request);
+  case IOCTL_SO_GETSOCKNAME:
+    return GetSockName(request);
+  case IOCTL_SO_GETPEERNAME:
+    return GetPeerName(request);
+  case IOCTL_SO_GETHOSTID:
+    return GetHostID(request);
+  case IOCTL_SO_INETATON:
+    return INetAToN(request);
```
Comment (leoetlino) [inline Source/Core/Core/IOS/Network/IP/Top.cpp]: Nitpicking, but the capitalisation looks a bit weird here. Could this be InetAToN instead?

File: Source/Core/Core/IOS/Network/IP/Top.cpp [hunk 3]
```diff
@@ -155,802 +155,844 @@ IPCCommandResult NetIPTop::IOCtl(const IOCtlRequest& request)
...
+    return GetHostByName(request);
+  case IOCTL_SO_ICMPCANCEL:
+    return ICMPCancel(request);
+  default:
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_NET);
     break;
   }
-  case IOCTL_SO_LISTEN:
+
+  return GetDefaultReply(0);
```
Comment (leoetlino) [inline Source/Core/Core/IOS/Network/IP/Top.cpp]: IPC_SUCCESS (this is a return code)

File: Source/Core/Core/IOS/Network/IP/Top.cpp [hunk 4]
```diff
@@ -155,802 +155,844 @@ IPCCommandResult NetIPTop::IOCtl(const IOCtlRequest& request)
...
     break;
   }
-  case IOCTL_SO_GETSOCKOPT:
-  {
-    u32 fd = Memory::Read_U32(request.buffer_out);
-    u32 level = Memory::Read_U32(request.buffer_out + 4);
-    u32 optname = Memory::Read_U32(request.buffer_out + 8);
 
-    request.Log(GetDeviceName(), LogTypes::IOS_WC24);
+  return GetDefaultReply(0);
```
Comment (leoetlino) [inline Source/Core/Core/IOS/Network/IP/Top.cpp]: IPC_SUCCESS
As a side note (probably for myself mainly :P), I doubt we should return success here. We should probably return IPC_EINVAL instead for unknown ioctlvs.

**Pull request merged**